File: frontend/lib/models/room.dart
class Room {
  final int id;
  final String code;
  final String building;
  final int floor;
  final String name;

  Room({required this.id, required this.code, required this.building, required this.floor, required this.name});

  factory Room.fromJson(Map<String, dynamic> json) => Room(
        id: json['id'],
        code: json['code'],
        building: json['building'],
        floor: json['floor'],
        name: json['name'],
      );
}

File: backend/app/models/room.py
from sqlalchemy import Column, Integer, String
from ..core.database import Base

class Room(Base):
    __tablename__ = 'rooms'

    id = Column(Integer, primary_key=True, index=True)
    code = Column(String, unique=True, index=True)
    building = Column(String)
    floor = Column(Integer)
    name = Column(String)
